v1/errors/sentry.go

```diff
@@ -0,0 +1,115 @@
+package errors
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	"gopkg.in/inconshreveable/log15.v2"
+)
+
+var (
+	// internalPackages represents a list of packages to be excluded from the errors stack trace sent to Sentry.
+	// Note: the Sentry SDK encodes dots in packages' import path into "%2e".
+	internalPackages = []string{
+		"github.com/exoscale/go-reporter",
+		"gopkg.in/inconshreveable/log15%2ev2",
+	}
+)
+
+// sentryEventModifier implements the sentry.EventModifier interface.
+type sentryEventModifier struct {
+	tags  map[string]string
+	panic bool
+}
+
+func (m *sentryEventModifier) ApplyToEvent(event *sentry.Event, _ *sentry.EventHint) *sentry.Event {
+	// Clean stack traces by filtering non-relevant frames
+	for i := range event.Exception {
+		event.Exception[i].Stacktrace.Frames = filterStackFrames(event.Exception[i].Stacktrace.Frames)
+	}
+	for i := range event.Threads {
+		event.Threads[i].Stacktrace.Frames = filterStackFrames(event.Threads[i].Stacktrace.Frames)
+	}
+
+	// Add tags extracted from the original log record context
+	event.Tags = m.tags
+
+	// If the event has been created following a panic, flag it as crashed
+	if m.panic && len(event.Threads) == 1 {
+		event.Threads[0].Crashed = true
+	}
+
+	return event
+}
+
+// sentryEventFromPanic returns a sentryEventModifier instance containing tags extracted from a log record's context.
+func sentryEventFromLogRecord(rec *log15.Record) *sentryEventModifier {
+	var s = sentryEventModifier{tags: make(map[string]string)}
+
+	// Extract context key/value pairs from the log record's context,
+	// which is stored as a string slice of key/value pairs.
+	for i := 0; i < len(rec.Ctx); i += 2 {
+		k, v := rec.Ctx[i], rec.Ctx[i+1]
```

Comment:
Possible to be out of range with the `i+1` in that case?

Reply:
@pierre-emmanuelJ good catch, it's indeed possible if someone records a log message with an odd number of context values... but this means we also have this bug in the current version of the reporter 😅 https://github.com/exoscale/go-reporter/blob/master/sentry.go#L28-L29

Reply:
Well, check the size can be useful ;)

Reply:
Yes, will do!

Reply:
Fixed in 0a0dff6.